### PR TITLE
Store merged notebooks in repo, remove dependency on Google Drive

### LINKED ISSUE #22: Document the GOOGLE_SERVICE_ACCOUNT environment variable
e.g. what to set it to.

### PR DESCRIPTION
**Pull request by duncandewhurst**
Closes #22 

This PR removes the dependency on Google Drive and the overly complex steps for [updating templates and adding components](https://github.com/open-contracting/notebooks-ocds#update-templates-with-new-components).

It also reorients the readme around the template notebooks, which are relevant to all users, rather than the component notebooks, which are only relevant to maintainers.

The trade-off is that maintainers might need to check out the repo to run `./manage.py pre-commit`,  but that command was already introduced when we added linting of SQL code.

If we think that it's important to enable maintenance without needing to check out the repo, we could consider using an action like https://github.com/ad-m/github-push-action to auto-update the templates when changes are made to the components, but I think this is fine for now.

Edit: Once merged, I can update the [notebooks on Google Drive](https://drive.google.com/drive/u/0/folders/1eb3pSQ55HylMsmwKqu7MrkvH12ROD4-9) to direct users back to the Github repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,51 @@
...
+[Structure and format feedback template](https://github.com/open-contracting/notebooks-ocds/blob/main/structure_and_format_feedback_template.ipynb) [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/open-contracting/notebooks-ocds/blob/main/structure_and_format_feedback_template.ipynb) | Use this notebook to provide feedback on structure and format issues reported by the Data Review Tool
+[Data quality feedback template](https://github.com/open-contracting/notebooks-ocds/blob/main/data_quality_feedback_template.ipynb) [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/open-contracting/notebooks-ocds/blob/main/data_quality_feedback_template.ipynb) | Use this notebook to provide detailed feedback on structure and format, conformance and coherence issues
```
Comment (jpmckinney) [inline README.md:12]:
```suggestion
[Data quality feedback template](https://github.com/open-contracting/notebooks-ocds/blob/main/data_quality_feedback_template.ipynb) [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/open-contracting/notebooks-ocds/blob/main/data_quality_feedback_template.ipynb) | Use this notebook to provide detailed feedback on structure, format, conformance and coherence issues 
```

File: README.md [hunk 2]
```diff
@@ -1,45 +1,51 @@
...
+## Contributing
 
-Alternatively, you can use the Open in Colab browser extension ([Chrome](https://chrome.google.com/webstore/detail/open-in-colab/), [Firefox](https://addons.mozilla.org/en-US/firefox/addon/open-in-colab/)) to add a button that, when clicked when viewing a Jupyter notebook on GitHub, will open that notebook in Colab.
+### Components
 
-## Templates
+To ease maintenance, the notebooks are made up of reusable components. The following table describes the relationship between the components and the notebooks.
 
-Each time a component is updated, GitHub Actions builds template notebooks from the components and writes them to a [Google Drive folder](https://drive.google.com/drive/u/0/folders/1eb3pSQ55HylMsmwKqu7MrkvH12ROD4-9).
+Component | Tasks | Used in
```
Comment (jpmckinney) [inline README.md:21]: I don't think the "Used in" column adds much that a contributor can't simply read in manage.py, and it adds maintenance burden to keep that column in sync. I suggest removing it.

File: README.md [hunk 3]
```diff
@@ -52,21 +58,16 @@ In Colab:
 1. Uncheck 'Include a link to Colaboratory'
 1. Select your branch, enter a commit message and click OK.
 
-### Update templates with new components
+### Add new components to a notebook
 
 For new components:
```
Comment (jpmckinney) [inline README.md:64]: This seems repetitive with the heading.

```suggestion
```

File: README.md [hunk 4]
```diff
@@ -52,21 +58,16 @@ In Colab:
...
 
-1. Add the component to the `nbmerge` commands in the 'Merge notebooks' section of [`main.yml`](https://github.com/open-contracting/kingfisher_notebook_components/blob/main/.github/workflows/main.yml) as required.
-2. Update the 'Components' and 'Templates' sections of `README.md`.
-
-### Add a new template
-
-1. Add a new step to the 'Merge notebooks' section of [`main.yml`](https://github.com/open-contracting/kingfisher_notebook_components/blob/main/.github/workflows/main.yml) with the `nbmerge` command to generate the template, e.g. `nbmerge component_1.ipynb component_2.ipynb > new_template.ipynb`.
-2. Upload an empty file to the [Kingfisher Notebooks Google Drive folder](https://drive.google.com/drive/u/0/folders/1eb3pSQ55HylMsmwKqu7MrkvH12ROD4-9) and get its Google Drive file ID from the 'Get link' menu option.
+1. Add the component to the entry for the notebook in the `NOTEBOOKS` variable in `manage.py`.
+2. Update the 'Components' table in `README.md`.
```
Comment (jpmckinney) [inline README.md:66]: Based on my earlier comment.

```suggestion
```

File: README.md [hunk 5]
```diff
@@ -1,45 +1,51 @@
...
-2. Add or edit cells, following the [style guide for SQL statements](https://ocp-software-handbook.readthedocs.io/en/latest/python/code.html#sql-statements). For new components, set a title using H2 formatting.
-3. Format any SQL code you add or edit, using [pg_format](https://github.com/darold/pgFormatter) with the following options: `pg_format -f 1`. Alternatively, you can use the [online version](https://sqlformat.darold.net/) and set the **Functions** option to 'lowercase'.
+### Edit a component
+
+1. Open the component in Colab.
+2. Add or edit cells, following the [style guide for SQL statements](https://ocp-software-handbook.readthedocs.io/en/latest/python/code.html#sql-statements).
+
+### Format your code
+
+1. Format any SQL code you add or edit, using [pg_format](https://github.com/darold/pgFormatter) with the following options: `pg_format -f 1`. Alternatively, you can use the [online version](https://sqlformat.darold.net/) and set the **Functions** option to 'lowercase'.
```
Comment (jpmckinney) [inline README.md:48]: In practice, would an analyst need to do this for each cell? It seems faster to commit the notebook to the branch, then run pre-commit locally.

File: README.md [hunk 6]
```diff
@@ -52,21 +58,16 @@ In Colab:
...
-2. Upload an empty file to the [Kingfisher Notebooks Google Drive folder](https://drive.google.com/drive/u/0/folders/1eb3pSQ55HylMsmwKqu7MrkvH12ROD4-9) and get its Google Drive file ID from the 'Get link' menu option.
+1. Add the component to the entry for the notebook in the `NOTEBOOKS` variable in `manage.py`.
+2. Update the 'Components' table in `README.md`.
 
-    ![Google Drive file ID](google_drive_file_id.png)
+### Add a new notebook
 
-3. Add a new entry to the `files` list in the [`upload`](https://github.com/open-contracting/kingfisher_notebook_components/blob/main/manage.py) function in `manage.py`, e.g. `('file_id', 'new_template.ipynb', 'New Template Title')`
+1. Add an entry for the the notebook and its components to the `NOTEBOOKS` variable in `manage.py`.
 4. Update the 'Templates' section of `README.md`.
```
Comment (jpmckinney) [inline README.md:71]:
```suggestion
4. Update the 'Notebooks' section of `README.md`.
```

**Review by jpmckinney — COMMENTED** [review]
I suppose the `GSA_CREDENTIALS` secret can be deleted in GitHub?

The "Create a branch" link now redirects to https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch

I'll do a little tidying in manage.py as a commit.

**Comment by jpmckinney** [comment]
> Edit: Once merged, I can update the [notebooks on Google Drive](https://drive.google.com/drive/u/0/folders/1eb3pSQ55HylMsmwKqu7MrkvH12ROD4-9) to direct users back to the Github repo

I assume the notebooks will be empty except for the instruction to go to GitHub?

File: README.md (hunk 2 above)
Comment (duncandewhurst) [reply README.md]: Done in https://github.com/open-contracting/notebooks-ocds/pull/30/commits/c584451718fab92970860c6ce2354ce2e4726972

File: README.md (hunk 5 above)
Comment (duncandewhurst) [reply README.md]: If editing many cells, yes. I've removed this section in https://github.com/open-contracting/notebooks-ocds/pull/30/commits/c584451718fab92970860c6ce2354ce2e4726972 in favour of the lining message which prompts the users to run pre-commit locally.

**Comment by duncandewhurst** [comment]
Thanks for those updates!

> I suppose the `GSA_CREDENTIALS` secret can be deleted in GitHub?

Yep, once this PR is merged.

> I assume the notebooks will be empty except for the instruction to go to GitHub?

Yep, that's the plan.

@yolile - happy for me to merge this whilst @jpmckinney is away? The only outstanding comments after applying his suggestions from the code review were about the readme, and I've addressed them in line with his feedback.

**Comment by yolile** [comment]
> @yolile - happy for me to merge this whilst @jpmckinney is away?

Sure!

**Comment by duncandewhurst** [comment]
I've updated the notebooks on Google Drive to redirect users to the versions in the repository.

@yolile I don't have access to the settings for this repository. Please can you delete the `GSA_CREDENTIALS` secret and add a branch protection rule to require status checks to pass before merging? Thanks!

**Comment by yolile** [comment]
> Please can you delete the GSA_CREDENTIALS

Done

> add a branch protection rule to require status checks to pass before merging?

I think this is already configured

**Comment by duncandewhurst** [comment]
Thanks!

**Pull request merged**